packages/aws-cdk-lib/aws-route53/test/record-set.test.ts

```diff
@@ -811,6 +811,60 @@ describe('record set', () => {
     });
   });
 
+  testDeprecated('Cross account zone delegation record with stsRegion', () => {
+    // GIVEN
+    const stack = new Stack();
+    const parentZone = new route53.PublicHostedZone(stack, 'ParentHostedZone', {
+      zoneName: 'myzone.com',
+      crossAccountZoneDelegationPrincipal: new iam.AccountPrincipal('123456789012'),
+    });
+
+    // WHEN
+    const childZone = new route53.PublicHostedZone(stack, 'ChildHostedZone', {
+      zoneName: 'sub.myzone.com',
+    });
+    new route53.CrossAccountZoneDelegationRecord(stack, 'Delegation', {
+      delegatedZone: childZone,
+      parentHostedZoneId: parentZone.hostedZoneId,
+      delegationRole: parentZone.crossAccountZoneDelegationRole!,
+      ttl: Duration.seconds(60),
+      removalPolicy: RemovalPolicy.RETAIN,
+      stsRegion: 'us-west-2',
```

Comment:
nit: can we use use a partition thats not a part of [the static list](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/custom-resource-handlers/lib/aws-route53/cross-account-zone-delegation-handler/index.ts#L108-L124) or a commercial region

Reply:
Changed this to a dummy region that doesn't exist. This test only asserts that it's serialized to the output of the CloudFormation, so it doesn't actually need to be a real region.